### PR TITLE
Remove some unnecessary initialization in string operations

### PR DESCRIPTION
**Pull request by AmjadHD**
* `prepareAdd`
* `toNimStr`
* `setLengthStrV2`
* `NimAsgnStrV2`
* `prepareMutation`
* Some cleanups

### CODE REVIEW

**Comment by juancarlospaco** [comment]
Add a descriptive comment maybe with a link to this PR about why it is the way you done it, so others dont break it in the future IMHO.

**Comment by AmjadHD** [comment]
The CI failure seems unrelated: `Error: cannot open 'test/tserialization.nim'` when testing the yaml package.

**Comment by AmjadHD** [comment]
@ringabout review please.

File: lib/system/strs_v2.nim [hunk 1]
```diff
@@ -34,53 +34,72 @@ template frees(s) =
...
+    cast[ptr NimStrPayload](reallocShared(p, contentSize(newLen)))
+  else:
+    cast[ptr NimStrPayload](realloc(p, contentSize(newLen)))
+
+template reallocPayload0(p: pointer; oldLen, newLen: int): ptr NimStrPayload =
+  when compileOption("threads"):
+    cast[ptr NimStrPayload](reallocShared0(p, contentSize(oldLen), contentSize(newLen)))
+  else:
+    cast[ptr NimStrPayload](realloc0(p, contentSize(oldLen), contentSize(newLen)))
+
```
Comment (Varriount) [inline lib/system/strs_v2.nim:60]: Do these *need* to be templates? I would think that any half-decent C compiler would recognize that these should be inlined.

Comment (Araq) [reply lib/system/strs_v2.nim:60]: The C compiler is already overwhelmed with getting all these inlines right and so I prefer the templates here.

File: lib/system/strs_v2.nim [hunk 2]
```diff
@@ -34,53 +34,72 @@ template frees(s) =
...
+      if newLen < newCap:
+        zeroMem(cast[pointer](addr s.p.data[newLen+1]), newCap - newLen)
 
 proc nimAddCharV1(s: var NimStringV2; c: char) {.compilerRtl, inl.} =
   #if (s.p == nil) or (s.len+1 > s.p.cap and not strlitFlag):
   prepareAdd(s, 1)
   s.p.data[s.len] = c
-  s.p.data[s.len+1] = '\0'
   inc s.len
+  s.p.data[s.len] = '\0'
```
Comment (Araq) [inline lib/system/strs_v2.nim:94]: That introduces a new read-after-write conflict for an out-of-order pipelined CPU. But most likely the optimizer will catch this.

**Pull request merged**